web/components/promtools-form.js

```diff
@@ -0,0 +1,254 @@
+                font-size: 1rem;
+                height: 1.5rem;
+                line-height: 1.5;
+            }
+            .input:hover {
+                border-color: var(--border-color-hover);
+            }
+            .input:focus {
+                outline: 0;
+                border-color: var(--primary-color);
+            }
+        `;
+    }
+
+    constructor() {
+        super();
+        this.target = 99.9;
+        this.metric = 'http_requests_total';
+        this.selectors = [{name: 'job', value: 'prometheus'}];
+
+        this.unavailabilityMinutes(this.target);
+    }
+
+    render() {
+        return html`
+            <form @submit="${this.generate}">
+                <div class="field">
+                    <label class="label" for="target">
+                        <h3>Availability (Unavailability in 30 days: ${this.unavailability})</h3>
+                        <div class="control">
+                            <input class="input" type="number" step="0.001" min="0" max="100" id="target"
+                                autofocus
+                                .value="${this.target}"
+                                @change="${(event) => this.unavailabilityMinutes(parseFloat(event.target.value))}"
+                                @keyup="${(event) => this.unavailabilityMinutes(parseFloat(event.target.value))}"
+                                ?disabled=${this.loading}
+                            />
+                        </div>
+                    </label>
+                    <p class="help">
+                        <a href="https://landing.google.com/sre/sre-book/chapters/availability-table/">Availability</a>
+                        is generally calculated based on how long a service was unavailable over some period.
+                    </p>
+                </div>
+                <div class="field">
+                    <label for="metric" class="label">
+                        <h3>Metric</h3>
+                        <div class="control">
+                            <input class="input" type="text" placeholder="http_requests_total" id="metric"
+                                autofocus
+                                .value="${this.metric}"
+                                @change="${(event) => this.metric = event.target.value}"
+                                ?disabled=${this.loading}
+                            />
+                        </div>
+                    </label>
+                    <p class="help">
+                        The metric name to base the SLO on.<br>
+                        It's best to base this on metrics coming from a LoadBalancer or Ingress.
+                    </p>
+                </div>
+
+                <div class="field group">
+                    <div style="flex: 1">
+                        <h3>Selectors</h3>
+                    </div>
+                    <div>
+                        <ui-button @click="${() => this.addSelector()}">+</ui-button>
```

Comment:
Can probably be simplified to `@click="${this.addSelector}"`

Reply:
That's correct, good catch :)